### ISSUE TITLE
Wrong error: <type_member> is not a pointee

### ISSUE THREAD

**Issue by clementval** [omni-compiler/omni-compiler#550]
The following code fails with `F_Front`

```fortran
MODULE mod1
  IMPLICIT NONE
  PRIVATE
  PUBLIC :: t_state

  TYPE, ABSTRACT :: t_state
    CLASS(t_State), POINTER :: parent => NULL()
  CONTAINS
    PROCEDURE :: init_state
  END TYPE t_State

CONTAINS
  SUBROUTINE init_state(this, that)
    CLASS(t_state), INTENT(inout), TARGET :: this
    CLASS(t_state), POINTER, OPTIONAL :: that 

    this%parent => that

    SELECT TYPE(that)
      CLASS IS (t_state)
        this%parent => that
    END SELECT

  END SUBROUTINE init_state
END MODULE mod1
```

The error is the following but the code is valid
```bash
"dummy.f90:mod1", line 21: 'that' is not a pointee.
```

Type of the pointer `that` seems to have wrong information in the `F-compile.c` -> `pointer_assignable` function. Looks like the `SELECT TYPE/CLASS IS` is causing the issue. If this code is outside of such a construct, the problem is not seen.
